gc.c

```diff
@@ -6819,7 +6825,27 @@ mark_const_tbl(rb_objspace_t *objspace, struct rb_id_table *tbl)
 #endif
 
 static void each_stack_location(rb_objspace_t *objspace, const rb_execution_context_t *ec,
-                                 const VALUE *stack_start, const VALUE *stack_end, void (*cb)(rb_objspace_t *, VALUE));
+                                 const VALUE *stack_start, const VALUE *stack_end, void *ctx,
+                                 void (*cb)(rb_objspace_t *, void *, VALUE));
+
+static void
+gc_mark_machine_stack_location_maybe(rb_objspace_t *objspace, void *ctx, VALUE obj)
```

Comment:
It'd be helpful for me if the intentional uses for `ctx` were documented in the sites where they were used. It's not immediately obvious what the intention of it is given that in the majority of callbacks it's unused.